cql3/statements/cas_request.cc

```diff
@@ -119,6 +119,9 @@ std::optional<mutation> cas_request::apply(foreign_ptr<lw_shared_ptr<query::resu
 
 const update_parameters::prefetch_data::row* cas_request::find_old_row(const cas_row_update& op) const {
     static const clustering_key empty_ckey = clustering_key::make_empty();
+    if (_key.empty()) {
+        throw exceptions::invalid_request_exception("partition key ranges empty - probably caused by an unset value");
```

Comment:
I think this is acceptable for branch 5.1, but please remember to also add such a check in master (maybe as on_internal_error), because if _key is empty because of some unknown bug, this code will crash - also on master.